### DISCUSSION TITLE
Unexpected 'size' Attribute Error in DE Marketplace

### DISCUSSION THREAD

**Discussion by bobyagnpf** [amzn/selling-partner-api-models#3653] [General]
Hi all,

I encountered an issue while creating a listing using /listings/2021-08-01/items/:sellerId/:sku. Specifically, the problem arose with the productType "SUIT" on the UK marketplace, where the marketplace_id is "A1F83G8C2ARO7P".

The schema I retrieved did not require nor include the 'size' attribute. Consequently, I was able to successfully submit the listing. However, I later discovered that the listing was blocked from being searchable. A call to listings/2021-08-01/item revealed the following issue:
```
"issues": [
    {
        "code": "90220",
        "message": "'size' is required but not supplied.",
        "severity": "ERROR",
        "attributeNames": [
            "size"
        ]
    }
],

```
When I checked the seller dashboard, it indicated that the 'size' attribute is mandatory. I have verified that the schema I used is the latest.

Can anyone confirm if this discrepancy between the schema and the seller dashboard is due to the schema not being updated in a timely manner? This has caused significant disruption to our product listing process. I appreciate your help in resolving this issue.

Looking forward to your replies.
Thanks!

![image](https://github.com/amzn/selling-partner-api-docs/assets/131427950/da0d3511-4d9c-4979-a58f-79f9148501f1)

**Comment by vaidas-lungis** [comment]
> ...The schema I retrieved did not require nor include the 'size' attribute. Consequently, I was able to successfully submit the listing.. 

Despite the successful submission, its only "accepted"

>  If accepted, initial validations have passed and the data has been submitted for further processing with the Amazon catalog  

According to the [Amazon Selling partner API docs Building Listings Management Workflows Guide](https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide) and last steps are checking `processing` (post submission) errors.  Preferably via the Notification API or just calling getListingItem after some time (this is not recommended and efficient way, but lets say quite easy to implement). 

![Screenshot 2023-07-20 at 10 58 00](https://github.com/amzn/selling-partner-api-docs/assets/5111555/b82fa267-c024-47b7-b0c8-ec1f97eb7136)


And answering your questions from my experience 

> Can anyone confirm if this discrepancy between the schema and the seller dashboard is due to the schema not being updated in a timely manner?

There are discrepancy and IMHO its not only due to timely manner, but most likely schema can't hold enough internal Amazon listing business logic therefore post submission errors are asynchronous 

Another good resource is [Guidance for Complex Attributes](https://developer-docs.amazon.com/sp-api/docs/listings-items-guidance-for-complex-attributes)

**Reply by bobyagnpf** [reply]
Thank you so much for your detailed response.
> According to the [Amazon Selling partner API docs Building Listings Management Workflows Guide](https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide) and last steps are checking (post submission) errors. Preferably via the Notification API or just calling getListingItem after some time (this is not recommended and efficient way, but lets say quite easy to implement).processing

Your suggestion to use 'getListingItem' to retrieve the listing status post-submission does indeed make a lot of sense, and it's something we've been adhering to.

As for the Notification API, we haven't subscribed to it yet. It seems like there might be some fees associated with SQS, correct me if I'm wrong.

> Another good resource is [Guidance for Complex Attributes](https://developer-docs.amazon.com/sp-api/docs/listings-items-guidance-for-complex-attributes)

We've successfully implemented many examples from the "Guidance for Complex Attributes" as long as the schema corresponds correctly with the seller dashboard.

> There are discrepancy and IMHO its not only due to timely manner, but most likely schema can't hold enough internal Amazon listing business logic therefore post submission errors are asynchronous

Therefore, I believe the issue we're experiencing might essentially be a bug. I suppose it would require the technical team at the Selling Partner API to resolve.

Looking forward to your continued assistance.

Best regards.
